components/notifications.js

```diff
@@ -77,6 +81,12 @@ function NotificationLayout ({ children, nid, href, as, fresh }) {
         router.push(href, as)
       }}
     >
+      {!!href && (
+        <Link
+          className={styles.linkBox}
+          href={typeof href === 'string' ? href : n.item?.parentId ? as + '?commentId=' + n.id : as}
```

Comment:
We want to hide the `?commentId` from the url for magicness.

`Link`'s `href` can be an object. https://nextjs.org/docs/pages/api-reference/components/link

I think we should be able to do `<Link href={{ pathname: href, asPath: as }} />`. You might've tried that though. The magicness isn't required but I like it personally.

Reply:
I think i resolved this too soon, I was fooled by the standard `router.push(href, as)` behavior but with the `<Link>` being primarily used for opening in a new tab (in my testing the `as` doesn't work when opening in a new tab) we should actually keep these query params

Reply:
We can keep the magicness when just regularly navigating but for opening in a new tab we will want these params

Reply:
That makes sense. I think the new window won't have the window state to read.